packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_request_wrapper.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/foundation.dart';
+import '../channel.dart';
+
+/// [SKProductRequestWrapper] wraps IOS SKProductRequest class to to retrive StoreKit product information in dart.
+///
+/// https://developer.apple.com/documentation/storekit/skproductsrequest?language=objc
+class SKProductRequestWrapper {
+  /// Get product list.
+  ///
+  /// [identifiers] is the product identifiers specified in Itunes Connect for the products that need to be retrived.
+  ///
+  /// Returns a future containing a list of [SKProduct] which then can be queried to get desired information.
+  static Future<List<SKProductWrapper>> getSKProductList(
+      List<String> identifiers) async {
+    final List<Map<dynamic, dynamic>> productListSerilized =
+        await channel.invokeListMethod<Map<dynamic, dynamic>>(
+      'getProductList',
+      <String, Object>{
+        'identifiers': identifiers,
+      },
+    );
+
+    final List<SKProductWrapper> productList = <SKProductWrapper>[];
+    for (Map<dynamic, dynamic> productMap in productListSerilized) {
+      productList.add(SKProductWrapper.fromMap(
+        productMap.cast<String, dynamic>(),
+      ));
+    }
+    return productList;
+  }
+}
+
+/// This class wraps the IOS SKProductSubscriptionPeriod class
```

Comment:
nits: iOS, period at the end of sentence, avoid the "This class" prefix.

I'd suggest an alternative phrasing in the spirit of:

```dart
/// A subscription period.
///
/// A period is defined by a [numberOfUnits] and a [unit], e.g for a 3 months period [numberOfUnits] is 3 and [unit] is a month.
///
/// This is a wrapper for StoreKit's [SKProductSubscriptionPeriod](https://developer.apple.com/documentation/storekit/skproductsubscriptionperiod?language=objc).
``` 

Same for the other Dartdocs below.